src/components/sbb-pearl-chain/sbb-pearl-chain.tsx

```diff
@@ -9,91 +10,141 @@ import legsData from './sbb-pearl-chain.helper';
 })
 export class SbbPearlChain {
   /**
-   * Define, if the pearl-chain represents a connection in the past,
-   * in the future or if it is a currently running connection.
-   * If it is currently running, provide a number between 0 and 100,
-   * which will represent the current location on the pearl-chain.
-   */
-  @Prop() public status?: InterfacePearlChainAttributes['status'];
-
-  /**
-   * Stringified JSON to define the legs of the pearl-chain.
+   * define the legs of the pearl-chain.
    * Format:
-   * `{"legs": [{"cancellation": true, "duration": 25}, ...]}`
-   * `duration`: number between 0 and 100. Duration of the leg is relative
+   * `{"legs": [{"duration": 25}, ...]}`
+   * `duration` in minutes. Duration of the leg is relative
    * to the total travel time. Example: departure 16:30, change at 16:40,
    * arrival at 17:00. So the change should have a duration of 33.33%.
-   * `cancellation`: if set, the leg will be marked as canceled.
    */
 
-  @Prop() public legs!: string;
+  @Prop() public legs!: InterfacePearlChainAttributes['legs'];
 
   /**
    * Per default, the current location has a pulsating animation. You can
    * disable the animation with this property.
    */
   @Prop() public disableAnimation?: boolean;
 
-  public render(): JSX.Element {
-    const legs = legsData(this.legs);
+  private _departureTime = this.legs && toDate(this.legs[0]?.departure?.time);
+  private _arrivalTime = this.legs && toDate(this.legs[this.legs?.length - 1].arrival?.time);
 
-    const statusClass = this.status === 'past' ? ' pearl-chain--past' : '';
+  private _getAllDuration(legs: InterfacePearlChainAttributes['legs']): number {
+    return legs.reduce((sum: number, leg) => (sum += leg.duration), 0);
+  }
 
-    let departureCancelClass = '';
-    let arrivalCancelClass = '';
+  private _getRelativeDuration(legs: InterfacePearlChainAttributes['legs'], leg: Leg): number {
+    const allDurations = this._getAllDuration(legs);
 
-    if (legs.length > 0) {
-      departureCancelClass = legs[0].cancellation ? ' pearl-chain--departure-cancellation' : '';
+    return Math.round((leg.duration / allDurations) * 100);
+  }
 
-      if (legs.length > 1) {
-        arrivalCancelClass = legs[legs.length - 1].cancellation
-          ? ' pearl-chain--arrival-cancellation'
-          : '';
-      }
+  private _getProgress(start: Date, end: Date): number {
+    const progress = differenceInMinutes(Date.now(), start);
+    const total = differenceInMinutes(end, start);
 
-      if (legs.length === 1) {
-        arrivalCancelClass = legs[0].cancellation ? ' pearl-chain--arrival-cancellation' : '';
-      }
+    return Math.round((progress / total) * 100);
+  }
+
+  private _getStatus(start: Date, end: Date): string {
+    if (isFuture(start)) {
+      return 'future';
+    }
+    if (isPast(start) && isPast(end)) {
```

Comment:
Start is in past, if end is in past, right?
if(isPast(end)) return 'past'

Consider using enum instead of string.